File: app/src/main/java/org/zwsmith/myapplication/core/dependencyInjection/AppModule.java
package org.zwsmith.myapplication.core.dependencyInjection;

import android.app.Application;

import org.zwsmith.myapplication.MyApplication;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {
    private MyApplication application;

    public AppModule(MyApplication application) {
        this.application = application;
    }

    @Provides
    @Singleton
    public MyApplication provideApplication() {
        return application;
    }
}

File: app/src/main/java/org/zwsmith/myapplication/core/dependencyInjection/AppModule.kt
package org.zwsmith.myapplication.core.dependencyInjection

import android.app.Application

import org.zwsmith.myapplication.MyApplication

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
class AppModule(private val application: MyApplication) {

    @Provides
    @Singleton
    fun provideApplication(): MyApplication {
        return application
    }
}
